### PR TITLE
optionally append to log

### PR DESCRIPTION
**Pull request by ghistes**
Hi,

this is just for me to learn the mechanics of creating pull-requests.

It's just a few lines to optionally append to the log-file, rather than overwriting it every time (I perfer it that way).

I don't really care whether or not you want this, but please let me know if there is any problem with the pull-request as such.

### CODE REVIEW

File: src/dbg.h [hunk 1]
```diff
@@ -49,12 +49,16 @@ static int xprintf(int fd, const char *fmt, ...)
 
 static int enabledbg(void)
 {
-	FILE *fp = fopen("/tmp/nnndbg", "w");
+        /* overwrite or append to previous log via environment */
+        int keep_log = getenv("NNN_DEBUG_KEEP_LOG") ? 1 : 0;
+        char *log_file_mode = keep_log ? "a" : "w";
```
Comment (KlzXS) [inline src/dbg.h]: Indentation is messed up. Please use tabs.

File: src/dbg.h [hunk 2]
```diff
@@ -68,7 +72,14 @@ static int enabledbg(void)
...
 
-	return 0;
+        if(keep_log) {
+           /* seperate the runs with a blank line if the file is not empty */
+           if(lseek(DEBUG_FD, 0, SEEK_CUR)) {  
+             write(DEBUG_FD, "\n", 1);
+           }
+        }
+
+        return 0;
```
Comment (KlzXS) [inline src/dbg.h]: Please fix Indentation.

File: src/dbg.h [hunk 3]
```diff
@@ -68,7 +72,14 @@ static int enabledbg(void)
 		return -1;
 	}
 
-	return 0;
+        if(keep_log) {
```
Comment (KlzXS) [inline src/dbg.h]: Use `if (keep_log) {`

File: src/dbg.h [hunk 4]
```diff
@@ -68,7 +72,14 @@ static int enabledbg(void)
...
+           /* seperate the runs with a blank line if the file is not empty */
+           if(lseek(DEBUG_FD, 0, SEEK_CUR)) {
```
Comment (KlzXS) [inline src/dbg.h]: Use `if lseek(DEBUG_FD, 0, SEEK_CUR)) {`. Mind the space.

**Comment by jarun** [comment]
How does appending help? To compare what happened in the last instance and current instance we will need to jump around. However, we can put our plugin `diffs` to good use ;). For a similar flow the logs would be similar and the diff would be very useful.

So i would suggest adding a timestamp to the log file. We can go to `/tmp` (or wherever the debug log is generated) and sort y time to get the logs from past runs in the correct order.

Also, if you do this and generate varying log names, when in debug mode, please dump the log filename to stderr (before entering curses mode) so we can just open the path in vim and start debugging.

**Comment by jarun** [comment]
Also, please rebase your view. It is outdated and conflicts with current code. `nnn` code changes very fast. Always rebase before raising a PR.

**Comment by jarun** [comment]
Also the timestamp should preferably in YYYYMMDDHHSS for simple sorting.

**Comment by jarun** [comment]
@ghistes any progress on this?

**Comment by jarun** [comment]
Closing as no response.

**Pull request closed without merging**